.github/workflows/ci.yml

```diff
@@ -0,0 +1,88 @@
+name: go-workflow
+on:
+  push:
+    # Temporarily removed for first PR
+    # branches:
+    #   - master
+  pull_request:
+jobs:
+  golang-tests:
+    name: tests
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    strategy:
+      matrix:
+        go-version: ["1.15", "1.14", "1.13"]
+        go111module: ["on", "off"]
+        include:
+          # includes goflags when go111module is on
+          - go111module: "on"
+            goflags: -mod=readonly
+    env:
+      GO111MODULE: ${{ matrix.go111module }}
+      GOPATH: ${{ github.workspace }}
+    # Required for building with GOPATH
+    # https://github.com/mvdan/github-actions-golang#how-do-i-set-up-a-gopath-build
+    defaults:
+      run:
+        working-directory: ${{ env.GOPATH }}/src/github.com/getsentry/sentry-go
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          path: ${{ env.GOPATH }}/src/github.com/getsentry/sentry-go
+          # Getting all history enables using `git merge-base origin/master HEAD`
+          fetch-depth: 0
+      - uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.go-version}}
+      - name: Remove if Go Module mode disabled
+        if: matrix.go111module == 'off'
+        run: |
+          # Iris does not supported in legacy GOPATH mode. We delete the source code
+          # because otherwise lint, build, and test steps would fail.
+          rm -vrf ./iris/ ./example/iris/
+          # go get is not required in Module mode
+          go get -v -t ./...
+      - name: Build
+        run: go build ./...
+      - name: Tests
+        run: |
+          go test ./...
+          go test ./... -race
+      - name: Run linting if Go Module is on
+        if: matrix.go111module == 'on'
```

Comment:
Over here I decided that we don't need to run linting for both modes. Please correct me if incorrect assumption.

Reply:
👍 

I think we've always run it because it did not have much impact on runtime, can't think now of a case where the assumption would be wrong.